### ISSUE TITLE
TypeScript failed to compile newer highlight API for v10.7.0

### ISSUE THREAD

**Issue by matsudai** [highlightjs/highlight.js#3060]
**Describe the issue**

I would like to use newer highlight API.
But TypeScript failed to compile.

**Which language seems to have the issue?**

**Are you using `highlight` or `highlightAuto`?**

I used `highlight`.

**Sample Code to Reproduce**

```ts
import hljs from 'highlight.js';
const code = '() => { console.log("A"); }';
const language = 'javascript';

// (1) Newer highlight.
// -> "Type error: Argument of type '{ language: string; }' is not assignable to parameter of type 'string'."
hljs.highlight(code, { language });

// (2) Deprecated highlight.
// -> Succeeded to compile But warn "Deprecated as of 10.7.0. Please use highlight(code, options) instead."
hljs.highlight(language, code);
```

**Expected behavior**

It worked when I modified `types/index.d.ts` as follows.

```ts
interface PublicApi {
    // highlight: (languageName: string, code: string, ignoreIllegals?: boolean, continuation?: Mode) => HighlightResult
    highlight: (codeOrlanguageName: string, optionsOrCode: string | HighlightOptions, ignoreIllegals?: boolean, continuation?: Mode) => HighlightResult
```

**Additional context**

Thank you and best regards,

**Comment by joshgoebel** [comment]
Is there some way we could add this to our linting check to automatically catch issues like this in the future?  Do you know what typescript console command we could use (despite the fact that our codebase is only hinted types, not full TS)?  I assume there is something since I get this type of feedback in VS Code live... so it should be possible to add this check to our lint CI to prevent things like this from passing CI in the future.

Thoughts?

**Comment by matsudai** [comment]
Sorry, I am JavaScript beginner. So this may be an irrelevant idea.
How about using [dtslint](https://github.com/microsoft/dtslint)? dtslint is used by [@types](https://github.com/DefinitelyTyped/DefinitelyTyped) as test tool.
dtslint tests a TypeScript declaration file. (but not execute funtions)

Thanks,

**Comment by joshgoebel** [comment]
Resolved by: https://github.com/highlightjs/highlight.js/commit/421b23b0ae15b55b4fd1a1b3187d5e6ba0c54cf8

@matsudai Thanks for report!

**Comment by joshgoebel** [comment]
Released 10.7.1.
